source/agora/common/EnrollmentManager.d

```diff
@@ -205,6 +205,78 @@ public class EnrollmentManager
         this.db.execute("DELETE FROM validator_set WHERE key = ?", enroll_hash[]);
     }
 
+    /***************************************************************************
+
+        In validatorSet DB, return the enrolled block height.
+
+        Params:
+            enroll_hash = key for an enrollment block height
+
+        Returns:
+            the enrolled block height.
+            Return size_t if no matching key exists.
+
+    ***************************************************************************/
+
+    public size_t getEnrolledHeight (const ref Hash enroll_hash) @trusted
+    {
+        try
+        {
+            auto results = this.db.execute("SELECT enrolled_height FROM validator_set" ~
+                " WHERE key = ?", enroll_hash[]);
+            if (!results.empty)
+                return results.oneValue!(size_t);
+            else
+                return size_t.init;
+        }
+        catch (Exception ex)
+        {
+            log.error("Database operation error {}", ex);
+            return size_t.init;
+        }
+    }
+
+    /***************************************************************************
+
+        Update the enrolled height of the validatorSet DB.
+
+        Params:
+            enroll_hash = enrollment blockheight to update enroll hash
+            block_height = enrolled blockheight
+
+        Returns:
+            Update execution status
+
+    ***************************************************************************/
+
+    public bool updateEnrolledHeight (const ref Hash enroll_hash,
+        const size_t block_height) @safe
+    {
+        size_t enrolled_hight = getEnrolledHeight(enroll_hash);
+        try
+        {
+            if (enrolled_hight > 0)
+            {
+                return false;
+            }
+            else
```

Comment:
I don't know what you mean.
Do you mean not to return in `if`?

Reply:
```D
if (foo())
{
    return false;
}
else
{
   some_code();
}
```

Is equivalent to:
```D
if (foo())
   return false;
some_code();
```

Except the later is usually more readable because the function flows more naturally and has less indentation.

Reply:
applied the suggestion.